native/screens/AccountSettings/UpdatePassword.jsx

```diff
@@ -0,0 +1,161 @@
+          newPassword: '',
+        });
+      }
+    }, 4000);
+  }, [toggleRerender]);
+
+  useEffect(() => {
+    const { currentPassword, newPassword, confirmNewPassword } = newPasswordForm;
+    if (currentPassword.length > 0) {
+      setTimeout(async () => {
+        const credentials = {
+          email,
+          password: currentPassword,
+        };
+        const res = await confirmPassword(credentials, userID); // pings API for every new character, once password security is enforced with minimum char amount, should restrict pinging this API to the minimum char #
+        if (res) {
+          setConfirmedPassword(true);
+          if (confirmNewPassword.length > 0 && newPassword === confirmNewPassword) {
+            setNewPasswordMatch(true);
+            setActiveUpdateButton(true);
+          } else {
+            setNewPasswordMatch(false);
+          }
+        } else {
+          setConfirmedPassword(false);
+        }
+        res ? setConfirmedPassword(true) : setConfirmedPassword(false);
+        if (newPassword === confirmNewPassword) {
+          setNewPasswordMatch(true);
+        } else {
+          setNewPasswordMatch(false);
+        }
+      }, 2000);
+    } else {
+      setConfirmedPassword(null);
+    }
+  }, [newPasswordForm]);
+
+  useEffect(() => {
+    if (!newPasswordsMatch && !confirmedPassword) {
+      setActiveUpdateButton(false);
+    } else {
+      setActiveUpdateButton(true);
+    }
+  }, []);
+
+  const handlePasswordUpdate = async () => {
+    const res = await updatePassword(newPasswordForm.newPassword, userID); // i feel that backend should also error check to confirm current password
+    if (res) {
+      setUpdateStatus('Success');
+      dispatch(uiActions.updateUser(res.user));
+      handleToggle(toggleRerender);
+    } else {
+      setUpdateStatus('Failed');
+    }
+  };
+
+  let UpdatePasswordStatusModal;
+  switch (updateStatus) {
+    case 'Success':
+      UpdatePasswordStatusModal = (
+        <Modal>
+          <View>
+            <Text>Password updated!</Text>
+          </View>
+        </Modal>
+      );
+      break;
+    case 'Failed':
+      UpdatePasswordStatusModal = (
+        <Modal>
+          <View>
+            <Text>Incorrect Password, try again</Text>
+            <Button title="OK" onPress={() => setUpdateStatus('Pending')} />
+          </View>
+        </Modal>
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  // this feels WET
+  return (
+    <View>
+      {updateStatus !== 'Pending' && UpdatePasswordStatusModal}
+      <Text>Confirm your password:</Text>
+      <TextInput
+        autoCapitalize="none"
+        returnKeyType="next"
+        secureTextEntry={true}
+        onChangeText={(currPassword) =>
+          handleTextChange(currPassword, 'currentPassword', setNewPasswordForm)
+        }
+        value={newPasswordForm.currentPassword}
+      />
+      {confirmedPassword !== null && <ConfirmedPasswordMessage status={confirmedPassword} />}
```

Comment:
I got an error when I tried to update a password; it looks like `ConfirmedPasswordMessage` hasn't been imported; easy fix